### ISSUE TITLE
Relocate migration guide to the Slicer wiki

### ISSUE THREAD

**Issue by jcfr** [Slicer/Slicer#6207]
## Is your feature request related to a problem? Please describe.

The `MigrationGuide` wiki page documents list of breaking API changes and how to update code to accommodate these.

See https://www.slicer.org/wiki/Documentation/Nightly/Developers/Tutorials/MigrationGuide

When possible, associated errors are also documented along with link to commit implementing the relevant changes.

Considering the wiki is in "life support" mode and not actively maintained, we need to ensure the information remains documented and discoverer-able using search engines.

## Describe the solution you'd like

The proposal is to keep a similar organization creating the following pages on the wiki:

```
Migration Guide
 |----- Slicer 5.0: API changes since 4.10
 |----- Slicer 4.10: API changes since 4.8
 |----- Slicer 4.8: API changes since 4.0
 |----- Slicer Extension updates
 |----- Obsolete Code Removal
 |----- Transition from ITK4 to ITK5
 |----- Transition from VTK 8.0 to VTK 9.0
 |----- Transition from VTK7-Qt4 to VTK8-Qt5
 |----- Transition from VTK6 to VTK7
 |----- Transition-from-VTK5 to VTK6
```

with the following differences:

* Organize changes with the section title `Slicer X.Y: API changes since X.Y-1` instead of specific `Slicer X.Y: Do this instead of that` top-level page for each change

* Consolidation of pages starting with `Slicer X.Y: ...` in relevant section

* Rename `Obsolete Code Removal` to `Code Modernization & Obsolete Code Removal`

### Page names

The page will **not** be prefixed with their parent page name.

This means the pages will have the following URLs and page names:

```
https://github.com/Slicer/Slicer/wiki/MigrationGuide
...
https://github.com/Slicer/Slicer/wiki/Slicer-4.10:-API-changes-since-4.8
...
https://github.com/Slicer/Slicer/wiki/Transition-from-VTK5-to-VTK6
```

and **NOT** these

```
https://github.com/Slicer/Slicer/wiki/MigrationGuide
...
https://github.com/Slicer/Slicer/wiki/MigrationGuide-Slicer-4.10:-API-changes-since-4.8
...
https://github.com/Slicer/Slicer/wiki/MigrationGuide-Transition-from-VTK5-to-VTK6
```

### Reference to modernization tools and scripts

These should be referenced in the `Code Modernization & Obsolete Code Removal` section.

## Describe alternatives you've considered

Add migration guide to readthedocs. 

## Additional context

NA

**Comment by jcfr** [comment]
To follow up on this comment:

> @jcfr Here are the scripts! 
> 
> [code_modernization_scripts.zip](https://github.com/Slicer/Slicer/files/8100340/code_modernization_scripts.zip)
> 
> Is there a directory in Slicer that would be a good place to share these and similar items?
> 
> Note: ITK has several of these scripts as well: ITK/Utilities/Maintenance
> 
> _Originally posted by @hjmjohnson in https://github.com/Slicer/Slicer/issues/6200#issuecomment-1045171387_

For now I suggest to add entries to the `Obsolete Code Removal` ton the Slicer [MigrationGuide](https://www.slicer.org/wiki/Documentation/Nightly/Developers/Tutorials/MigrationGuide#Obsolete_Code_Removal) managed as mediawiki pages.

For organization of the scripts, we should reference the following location and avoid duplication:

| Context | URL |
|--|--|
| Slicer | https://github.com/Slicer/Slicer/tree/master/Utilities/Scripts |
| ITK | https://github.com/InsightSoftwareConsortium/ITK/tree/master/Utilities/Maintenance |
| VTK | https://github.com/Kitware/VTK/tree/master/Utilities/Maintenance |
| CMake | https://github.com/jcfr/cmake-auto-style |

> re: CMake

I would have no  problem relocation the `cmake-auto-style` to a "better" location.

**Comment by jcfr** [comment]
> You all still want to have the Migration Guide on the slicer.org/wiki (https://www.slicer.org/wiki/Documentation/Nightly/Developers/Tutorials/MigrationGuide) or should it be migrated to the GitHub repository like https://github.com/InsightSoftwareConsortium/ITK/blob/master/Documentation/ITK5MigrationGuide.md considering developers are likely the individuals writing the updates?

_Originally posted by @jamesobutler in https://github.com/Slicer/Slicer/issues/6750#issuecomment-1447463916_

**Comment by jcfr** [comment]
> You all still want to have the Migration Guide on the slicer.org/wiki 

The plan originally suggested above was to keep a similar organization creating the following pages on the GitHub wiki.

Since the documentation is specific to a given version of Slicer, maintaining these on the wiki is sensible.

As documented [here](https://docs.github.com/en/communities/documenting-your-project-with-wikis/changing-access-permissions-for-wikis), we could also restrict edit access:

![image](https://user-images.githubusercontent.com/219043/221738482-88f3a88a-44c5-4dea-ae11-bd0f6f0ac100.png)

**Comment by jcfr** [comment]
> Based on the ordering, would a new Slicer 5.3 prefixed entry be placed at position 1.2 as the most recent 5.3 change? Is the 1.1 Slicer 5 changes since 4.10 section still needing to be first entry instead of chronologically mentioned?
> 
> <img width=50% src="https://user-images.githubusercontent.com/15837524/221735251-8c063aee-ee5d-4411-a368-04fd78efd360.png"/>

_Originally posted by @jamesobutler in https://github.com/Slicer/Slicer/issues/6750#issuecomment-1447463916_

**Comment by jcfr** [comment]
> re: Slicer 5.3

The ordering has just been fixed.

| Before | After |
|--|--|
|![image](https://user-images.githubusercontent.com/15837524/221735251-8c063aee-ee5d-4411-a368-04fd78efd360.png) | ![image](https://user-images.githubusercontent.com/219043/221739383-0d3ecbf2-b22d-4f84-844c-ccedbf30fcf1.png) |

**Comment by lassoan** [comment]
Yes, moving to the Slicer Github wiki would be a good place. And yes, we should restrict editing so that contributions need to go through the pull request process.
